InscryptionAPI/Saves/ModdedSaveData.cs

```diff
@@ -42,16 +78,32 @@ public bool GetValueAsBoolean(string guid, string key)
         return result;
     }
 
-    public void SetValue(string guid, string key, object value)
+    /// <summary>
+    /// Set the value of a key as an object in the save data,
+    /// It's recommended to not save an object that implements Unity's Object class as it can cause a infinite recursion and crash the game.
+    /// </summary>
+    /// <param name="guid">The GUID of the mod.</param>
+    /// <param name="key">The key to set the value of.</param>
+    /// <param name="value">The object value to set.</param>
+    public void SetValueAsObject(string guid, string key, object value)
```

Comment:
I'm a bit concerned about having two SetValues that behave differently (one just toStrings the object and one stores it as-is). I do realize you're constrained by the old design of the save manager that only allowed strings, ints, floats, and bools.

What I would suggest is that you just keep a single SetValue method, but check the type of the object to determine how you store it. If it's a string, int, float, or bool (or if the type IsPrimitive, just to cover some bases), then store the tostring() of the object - otherwise, store the object itself.

This way there's still just one SetValue and we eliminate potential confusion from the user.

Reply:
This was done purely for complatibility issue, Not constrained, In fact, If i were to remove the `SetValue` and rename `SetValueAsObject` to `SetValue` the code will work, but break complatibility (Both InscryptionAPI and other mod)

Reply:
I understand that it's for compatibility. However, my concern is that they both have the same parameters and similar names, potentially leading to confusion:

```c#
public void SetValue(string guid, string name, object value);

public void SetValueAsObject(string guid, string name, object value);
```

Yes, the docs tell you the difference, but if at all possible we should make it obvious in the naming of the methods what they do. The original 'SetValue' accepted 'object' for convenience - at the time we didn't consider saving whole objects to the save file. This PR adds a lot of value by allowing that, but this leads to the potential for confusion.

My suggestion is either something like this (pseudo-code-ish):

```c#

public void SetValue(string guid, string name, object value)
{
  Type vType = value.GetType()
  if (vType  == typeof(float) || vType == typeof(string) || vType.IsPrimitive())
    // follow the old logic where the value is stringified and saved
  else
    // follow the new logic where the value is saved as an object in the save file

}
```

Or, if you would rather have an explicit method, consider being even more explicit:

```c#
public void SetValue(string guid, string name, object value);

public void SetObject<T>(string guid, string name, T value);
```

Reply:
I have choose the latter option, Because,
Primative type is not the only thing that got `.ToString()`, Some people also pass object, which got `.ToString()` and if we only check for primitive type, we won't be able to catch that object.